### DISCUSSION TITLE
onSubmit takes long time, and show loader when isSubmitting dealy

### DISCUSSION THREAD

**Discussion by dridi93** [react-hook-form/react-hook-form#8001] [Q&A]
Hi,

I am not sure if it's a bug or not,
but when i use isSubmitting to display the loader it takes time to appear,
I think it's releated to "handleSubmit",

**Code example**

package.json
`
    "react": "17.0.2",
    "react-native": "0.67.3",
    "react-hook-form": "7.27.1",
    "native-base": "3.3.7", // UI Lib
`

form.tsx

`
  const {control, handleSubmit, reset, watch, setValue, getValues} = useForm();
  const {isSubmitting, isSubmitSuccessful} = useFormState({control});
  ...
  const _onSubmit = (item: any) => {console.log('Data: ', item)};
  ...

    <Button
      size={'lg'}
      colorScheme={'success'}
      isLoading={isSubmitting}
      onPress={handleSubmit(_onSubmit)}>
      Valider
    </Button>
`

**Screenshot**

![screencast-nimbus-capture-2022 03 10-11_51_43](https://user-images.githubusercontent.com/11330682/157647845-142650a1-de37-4571-820a-56b69c0e3d82.gif)

**Expected behavior**

Submit the form fast, and the loader appear immediatly,

**Comment by Moshyfawn** [comment]
I don't think we can help you without a proper reproduction of the use-case you have.

I've put together a very basic [RN Snack](https://snack.expo.dev/@moshyfawn/nb-rhf) with RHF integration with Native Base and I couldn't reproduce the issue.

Please, take a look at the [RN Snack](https://snack.expo.dev/@moshyfawn/nb-rhf) and consider tweaking it to match your use-case better.
